src/lib/global.ts

```diff
@@ -17,7 +21,8 @@ export interface GlobalMutations {
   loadLangData(): void
 }
 export class Global {
-  data = reactive({ status: { state: "loading", message: "loading" }, loc: { selected: 'en', text, }, client, launchOnBoot })
+  private client!: ClientType
+  data = reactive({ status: { state: "loading", message: "loading" }, loc: { selected: 'en', text }, launchOnBoot: new AutoLauncher, client: this.client })
```

Comment:
I do not understand what this does either.

You just moved `client` variable to private property that is incorrectly typed as `ClientType` (there is no private constructor to enforce this invariant and no methods in the class that assign anything to the field in general), in fact it is guaranteed to always be `undefined` and nothing else.

Reply:
Hm, I think I do understand. You want to have async initialization, but you also want to pretend it is initialized synchronously for convenient use elsewhere. But it will result in strange bugs because the invariant you rely on are can't be guaranteed and you have to call `init()` for it to actually be true. Am I right?

Reply:
Hm... it must already be broken, `this.initClient()` in `init()` is commented-out, so there is no code in this file that can guarantee `data.client` is actually anything other than `undefined`. Does it actually work?

Reply:
I can't initialize it in the constructor because it's async. I have to have some reference to it inside data for other code to be valid. `this.initClient()` is commented out for debug purposes. I appreciate the scrutiny but in this case we are nitpicking code that isn't meant to be close to finalized. Once we merge the latest branches together I will refactor Client into a class in which case it can be initialized using `new Client` similar to how I handle `launchOnBoot`.

Reply:
If client isn't used, let's remove it from `data`. Otherwise as far as I understand it this is guaranteed to be broken, I don't see a working code path.

As to async constructors, you can have them:
```typescript
class T {
  private construct(public readonly client) {}
  public static async init(): Promise<T> {
    const client = await Client.init();
    return new T(client);
  }
}
```

This way you physically can't construct invalid invariant of `T` because its constructor is private and the only way to ever get its instance is to go through `T.init()`.